sovtokenfees/sovtokenfees/req_handlers/fees_utils.py

```diff
@@ -0,0 +1,53 @@
+from sovtokenfees.domain import build_path_for_set_fees
+
+from common.serializers.serialization import state_roots_serializer, proof_nodes_serializer, config_state_serializer
+from plenum.common.constants import BLS_LABEL, MULTI_SIGNATURE, ROOT_HASH, PROOF_NODES
+from state.trie.pruning_trie import rlp_decode
+
+
+class BatchController:
+    def __init__(self):
+        self.fees_in_current_batch = 0
+        self._deducted_fees = {}
+
+    def add_deducted_fees(self, txn_type, seq_no, fees):
+        key = "{}#{}".format(txn_type, seq_no)
+        self._deducted_fees[key] = fees
+
+    def has_deducted_fees(self, txn_type, seq_no):
+        return "{}#{}".format(txn_type, seq_no) in self._deducted_fees
+
+
+def get_fee_from_state(state, fees_alias=None, is_committed=False, with_proof=False, bls_store=None):
+    fees = None
+    proof = None
+    try:
+        fees_key = build_path_for_set_fees(alias=fees_alias)
+        if with_proof:
+            proof, serz = state.generate_state_proof(fees_key,
```

Comment:
It doesn't take into account `is_committed` flag, so that it always returns uncommitted value.
Please add tests for this (send SET_FEEs, but delay COMMITs so that the batch is not ordered (stay uncommitted); send GET_FEE; make sure that committed value is returned).
Please have a look at `get_value_from_state` in `LedgerRequestHandler`.

Reply:
done